### ISSUE TITLE
On the performance of FIFA bot

### ISSUE THREAD

**Issue by Maphist0** [ChintanTrivedi/DeepGamingAI_FIFA#3]
Great work and thank you for sharing this interesting FIFA bot !!
I'm wondering whether the goal in the Youtube video (linked in the README) is performed by the agent?

I run your code on my setup, but the performance is somehow not ideal. Some strange things are observed:
  1. A lot of subtle movements are generated by the movement LSTM branch. The player literally won't move towards the opponent's goal.
  1. The action LSTM network produces, in most cases, the no-op option. (I guess the last empty entry in [this variable](https://github.com/ChintanTrivedi/DeepGamingAI_FIFA/blob/master/play_fifa.py#L72) stands for no-op, am I right?)

What's your experience on improving the performance? Thank you ahead of time!

**Comment by ChintanTrivedi** [comment]
Yes, the bot behaves bonkers for the most part, and I need to keep it
running for few hours to get some magic moments out of it. This is a huge
limitation of supervised learning setup, where the bot doesn't really
understand the game but only tries to imitate based on what it has seen
previously in training. Fifa being such an open and varied game, there are
very few instances that will be well covered in the training phase. Rest of
the instances the bot won't know what to do since it hasn't been trained
enough.

It was trained under a few fixed circumstances, like always going from left
to right, always chelsea (blue) vs red (arsenal or manutd) and always with
game running at 720p windowed with lowest graphic settings. Unlike humans,
neural nets are easily thrown off by subtle changes in input distribution.
So for best results, continue training process of the bot under your own
settings so when you test it, the input is exactly the same as the input
you gave it during training.

Apart from that, it's still an open research problem with OpenAI doing the
most promising work in making AI bots for games. Best way to improve is to
follow their work and see what can be implemented with the restriction of
not having access to the game's code.

Good luck and thanks for reaching out.

Chintan
